google/resource_compute_instance_template_migrate.go

```diff
@@ -0,0 +1,50 @@
+package google
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func resourceComputeInstanceTemplateMigrateState(
+	v int, is *terraform.InstanceState, meta interface{}) (*terraform.InstanceState, error) {
+	if is.Empty() {
+		log.Println("[DEBUG] Empty InstanceState; nothing to migrate.")
+		return is, nil
+	}
+
+	switch v {
+	case 0:
+		log.Println("[INFO] Found Compute Instance Template State v0; migrating to v1")
+		return migrateComputeInstanceTemplateStateV0toV1(is)
+	default:
+		return is, fmt.Errorf("Unexpected schema version: %d", v)
+	}
+}
+
+func migrateComputeInstanceTemplateStateV0toV1(is *terraform.InstanceState) (*terraform.InstanceState, error) {
+	log.Printf("[DEBUG] Attributes before migration: %#v", is.Attributes)
+
+	// automatic_restart is stored in two places. The top-level automatic_restart value is deprecated, so let's delete
+	// it from the state map for now. For paranoia's sake, we compare it to the value stored in scheduling as well.
+	ar := is.Attributes["automatic_restart"]
+	delete(is.Attributes, "automatic_restart")
+
+	if is.Attributes["scheduling.#"] != "1" {
```

Comment:
It is optional, but it is always written to the state file (as we always read a value).